### ISSUE TITLE
Protocols

### ISSUE THREAD

**Issue by Aehmlo** [Aehmlo/deoxy#10]
Implement protocols (bikeshed that name) that allow specifying a sequence of actions to take at specified intervals in order to string together an entire program.

The implementation for this will probably look something like the following, but more elegant.
```rust
enum Action {
    // Roughly 1:1 correspondence with messages that can be sent, with associated motors/pins.
}
struct Program {
    steps: Vec<Action>,
}
impl Iterator for Program {
    type Offset = Duration;
    type Item = (Offset, Action);
}
```
This would then be easy to
1) display to the user in a logical format, and
2) schedule using `ScheduledAction`, since the delay would be built into the iterator.

**Comment by Aehmlo** [comment]
I would also like to implement a trait that would be useable to send these just like `Message`s.

**Comment by Aehmlo** [comment]
Closed in 711f462.
